test/jdk/java/nio/channels/Selector/DeferredCloseTest.java

```diff
@@ -0,0 +1,573 @@
+        // so that the close is deferred by the channel implementations.
+        System.out.println("closing channel: " + channel);
+        assertTrue(channel.isRegistered(), "channel isn't registered: " + channel);
+        channel.close();
+        // wait for the operation on the channel and the selector task to complete
+        channelOpResult.get();
+        selectorTaskResult.get();
+    }
+
+    /*
+     * Keeps invoking Selector.select() until the channel is closed, after which
+     * it cancels the SelectionKey and does one last Selector.select() to finish
+     * the deferred close.
+     */
+    private static final class SelectorTask implements Callable<Void> {
+        private final SelectionKey selectionKey;
+        private final CountDownLatch startedLatch;
+
+        private SelectorTask(SelectionKey selectionKey, CountDownLatch startedLatch) {
+            this.selectionKey = Objects.requireNonNull(selectionKey);
+            this.startedLatch = startedLatch;
+        }
+
+        @Override
+        public Void call() throws Exception {
+            try {
+                // notify that the task has started
+                startedLatch.countDown();
+                Selector selector = selectionKey.selector();
+                SelectableChannel channel = selectionKey.channel();
+                while (true) {
+                    selector.select(10);
+                    if (!channel.isOpen()) {
+                        // the channel is (defer) closed, cancel the registration and then
+                        // issue a select() so that the Selector finishes the deferred
+                        // close of the channel.
+                        System.out.println("channel: " + channel + " isn't open," +
+                                " now cancelling key: " + selectionKey);
+                        selectionKey.cancel();
+                        System.out.println("initiating select after key cancelled: " + selectionKey);
+                        selector.select(5);
+                        break;
+                    }
+                }
+            } catch (ClosedSelectorException _) {
+            }
+            return null;
+        }
+    }
+
+    private static final class ConnAcceptor implements Callable<SocketChannel> {
+        private final ServerSocketChannel serverSocketChannel;
+
+        private ConnAcceptor(ServerSocketChannel serverSocketChannel) {
+            this.serverSocketChannel = serverSocketChannel;
+        }
+
+        @Override
+        public SocketChannel call() throws Exception {
+            SocketChannel accepted = serverSocketChannel.accept();
+            System.out.println("Accepted connection: " + accepted);
+            return accepted;
+        }
+    }
+
+
+    private static final class DelayInjectingNIOAccess implements JavaNioAccess {
+        private final JavaNioAccess realJavaNioAccess = SharedSecrets.getJavaNioAccess();
```

Comment:
We can't have tests implementing this interface as it puts on a tax on every change to this interface (it changes regular). So I think we need to drop this and come up with an another way to test this scenario.